src/pages/my_page/user_profile/MyProfile.tsx

```diff
@@ -1,46 +1,177 @@
 import { useTranslation } from 'react-i18next';
-import { useSelector } from 'react-redux';
-import { Box, Divider } from '@mui/material';
+import { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
 import { Helmet } from 'react-helmet-async';
-import { RootState } from '../../../redux/store';
-import { UserInfo } from './UserInfo';
+import { Box, IconButton, TextField, Tooltip, Typography } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import EditIcon from '@mui/icons-material/Edit';
+import { ErrorMessage, Field, FieldProps, Form, Formik, FormikProps } from 'formik';
 import { UserOrcid } from './UserOrcid';
 import { UserRoles } from './UserRoles';
-import { BackgroundDiv } from '../../../components/styled/Wrappers';
 import { ResearchProfilePanel } from './ResearchProfilePanel';
+import { RootState } from '../../../redux/store';
+import { setNotification } from '../../../redux/notificationSlice';
+import { BackgroundDiv, InputContainerBox } from '../../../components/styled/Wrappers';
+import { PageSpinner } from '../../../components/PageSpinner';
+import { CristinPerson, FlatCristinPerson } from '../../../types/user.types';
+import { updateCristinPerson } from '../../../api/userApi';
+import { useFetch } from '../../../utils/hooks/useFetch';
+import { UrlPathTemplate } from '../../../utils/urlPaths';
+import { getValueByKey } from '../../../utils/user-helpers';
+import { isErrorStatus, isSuccessStatus } from '../../../utils/constants';
 
 export const MyProfile = () => {
   const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const location = useLocation();
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   const user = useSelector((store: RootState) => store.user)!; // If user has been empty this route would already be blocked
 
+  const currentCristinId = useSelector((store: RootState) => store.user?.cristinId) ?? '';
+  const isPublicPage = location.pathname === UrlPathTemplate.ResearchProfile;
```

Comment:
MyProfile er vel bare for innlogget bruker, hvor man kan se sin egen data, så denne blir alltid `false` eller?

Reply:
Stemmer. Det hang nok igjen fra når Research Profile Panel ble opprettet, og det var snakk om å bruke samme komponent på begge sidene.